allennlp/training/trainer.py

```diff
@@ -328,10 +328,10 @@ def __init__(
         self,
         model: Model,
         optimizer: torch.optim.Optimizer,
-        data_loader: torch.utils.data.DataLoader,
```

Comment:
There are small parts of the guide that should be updated with this change.  Not sure how to keep this in sync with the guide / when to release things.

Reply:
I feel like the guide should be targeted at the latest stable version (1.0.0 right now), right? But it would definitely be good to have some CI for the guide. I don't know how hard that would be.

Reply:
Does this still refer to the pytorch `DataLoader` class? It should if we want this trainer to work with arbitrary `DataLoader`s.

Reply:
This refers to the new general base class

Reply:
So the trainer won't work with non-AllenNLP data loaders? That is a problem, right?

Reply:
It will work with any class that implements `allennlp.data.DataLoader`, such as `allennlp.data.PyTorchDataLoader`. So this is more general than before.